### ISSUE TITLE
1.16.5 Spigot - ES System is missing texture and throws a null exception. (Partially independently fixed: Still has Issue #11)

### ISSUE THREAD

**Issue by nidanninja** [SeanOMik/EnergeticStorage#18]
Hello,

As mentioned on a few other issues, Snapshot 0.5.1 is not functioning on 1.16.5 Spigot. I was able to get Snapshot 0.5 working on 1.16.5 Spigot, though the previous bugs were still present.

Previously placed ES System player heads from Snapshot 0.5 are still functional with Snapshot 0.5.1, but revert to a normal player head when broken, losing all NBT data.

ES Systems acquired through /esgive do not have a texture, and are non-functional when placed anywhere. Any normally crafted ES Systems also lack the texture, and appear as a normal player head. Advanced tooltips shows in both of these cases that the player head has 3 NBT tags, though ES Systems created in Snapshot 0.5 have 4 NBT tags (presumably the texture being the 4th, and missing, tag).

I looked through the raw code + the error messages, and with as much information as I have I'd guess the issue lies with applying the texture to the Computer skull. It seems that Skulls.java never applies the texture, then Utils.java runs isBlockASystem(), returning null when it compares the texture (as there is no texture applied). 

From what I saw looking through the Git Blame / History, I think the fix *might* be on Line 24 of Skulls.java, as the previous version was:
`item = createSkull(texture, name); `
while the current version says:
`item = createSkull(uuid, name); `

It appears that this could lead to the texture not being added. Furthermore, the `String url` variable is not used in createSkull (except for the check if it is empty), meaning the UUID being passed into the method is not being used - could this be the cause? The previous version of Skulls.java contained this line using the url variable, which is now missing:
`gameProfile.getProperties().put("textures", new Property("textures", url)); `

Could the above be related to line 42 of Skulls.java, `ownerNBT.setString("Id", uuid); `, and potentially replacing `uuid` with `url` so it uses the argument instead of the class variable to fix the issue?

I'm not 100% sure that any of these are the fix, but it's my best guess with the information I have - I've not done any Java in a while. Hopefully some of this information helps get this fixed - I really love this plugin and want it to work! I've also attached the error messages when placing and breaking the ES System.

[ES_Break.txt](https://github.com/SeanOMik/EnergeticStorage/files/6216911/ES_Break.txt)
[ES_Place.txt](https://github.com/SeanOMik/EnergeticStorage/files/6216912/ES_Place.txt)

**Comment by SeanOMik** [comment]
I just updated the plugin for 1.16.5 [here](https://github.com/SeanOMik/EnergeticStorage/releases/tag/0.6.0-SNAPSHOT). I'll take a look at #11

**Comment by SeanOMik** [comment]
I'll take a look at #11. Could you show your code for updating to 1.16.5?

**Comment by SeanOMik** [comment]
Actually, nvm, I don't the code for 1.16.5, I just [updated it here](https://github.com/SeanOMik/EnergeticStorage/releases/tag/0.6.0-SNAPSHOT) for 1.16.5.

**Comment by nidanninja** [comment]
Hey, just saw this.

I simply rewrote the createSkull() method in Skulls.java to fix the NBT data issue, using com.mojang.authlib (as before), and removing the NBT API dependency. I just figured it was easier for me than learning to use the NBT API dependency, since the old code with authlib worked fine. It now properly applies the texture and has all the proper SkullOwner NBT data, so it passes the isBlockASystem check.

As for Issue #11 , I just ended up making a work-around by only allowing shift+click for drives, and it seems to have stopped the issue entirely. I'm not sure exactly what the issue was, but something with putting a drive in repeatedly with the left click would duplicate it in the List, so I just disabled the main click functionality (and immediately cancel the input when attempted), and only allow shift+click to add Drives (in ESDriveGUI.java).

I can still upload my changes if you'd like them, just let me know. I'll also be sure to test your new snapshot soon to see if it works fine on my server.

Thanks!
